ckanext/qa/tasks.py

```diff
@@ -356,35 +369,147 @@ def score_by_sniffing_data(archival, resource, score_reasons):
+                score_reasons.append(_('A system error occurred during downloading this file') + '. %s' % e)
                 return (None, None)
         else:
-            # No cache_url
-            if archival.status_id == Status.by_text('Chose not to download'):
-                score_reasons.append(_('File was not downloaded deliberately') + '. '
-                                     + _('Reason') + ': %s. ' % archival.reason + _('Using other methods to determine file openness.'))
-                return (None, None)
-            elif archival.is_broken is None and archival.status_id:
-                # i.e. 'Download failure' or 'System error during archival'
-                score_reasons.append(_('A system error occurred during downloading this file') + '. '
-                                     + _('Reason') + ': %s. ' % archival.reason + _('Using other methods to determine file openness.'))
-                return (None, None)
-            else:
-                score_reasons.append(_('This file had not been downloaded at the time of scoring it.'))
-                return (None, None)
+            score_reasons.append(_('Cache filepath does not exist: "%s".') % filepath)
+            return (None, None)
+    if filepath:
+        sniffed_format = sniff_file_format(filepath)
+        if delete_file:
+            try:
+                os.remove(filepath)
+            except OSError as e:
+                log.warn("Unable to remove temporary file %s: %s", filepath, e)
+        score = lib.resource_format_scores().get(sniffed_format['format']) \
+            if sniffed_format else None
+        if sniffed_format:
+            score_reasons.append(_('Content of file appeared to be format "%s" which receives openness score: %s.')
+                                 % (sniffed_format['format'], score))
+            return score, sniffed_format['format']
+        else:
+            score_reasons.append(_('The format of the file was not recognized from its contents.'))
+            return (None, None)
+    else:
+        # No cache_url
+        if archival.status_id == Status.by_text('Chose not to download'):
+            score_reasons.append(_('File was not downloaded deliberately') + '. '
+                                 + _('Reason') + ': %s. ' % archival.reason + _('Using other methods to determine file openness.'))
+            return (None, None)
+        elif archival.is_broken is None and archival.status_id:
+            # i.e. 'Download failure' or 'System error during archival'
+            score_reasons.append(_('A system error occurred during downloading this file') + '. '
+                                 + _('Reason') + ': %s. ' % archival.reason + _('Using other methods to determine file openness.'))
+            return (None, None)
+        else:
+            score_reasons.append(_('This file had not been downloaded at the time of scoring it.'))
+            return (None, None)
+
+
+def _download_url(url):
```

Comment:
could the download url code be in a helper class?

Reply:
It could, if there was a good case for sharing it with other code. At this point that doesn't really seem practical.